Horizontal read scaling

atovpeko opened the pull request:
Closes https://linear.app/timescale/issue/ORC-2/docs

atovpeko committed:
draft updated

atovpeko committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

billy-the-fish commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -8,71 +8,110 @@ tags: [replicas, scaling, ha]
-replicas have minimal impact on the performance of the primary data instance. This enables you to interact with 
-up-to-date production data for analysis or to scale out reads beyond the limits of your primary data instance. You use 
-read replicas for read scaling. To limit data loss for your Timescale Cloud services, use [High availability][ha].
+A $READ_REPLICAs is a read-only copy of the primary data instance in your $CLOUD_LONG. Queries on $READ_REPLICAs have minimal impact on the performance of the primary data instance. This enables you to interact with 
```
```suggestion
A $READ_REPLICA is a read-only copy of the primary data instance in your $SERVICE_LONG. Queries on $READ_REPLICAs have minimal impact on the performance of the primary data instance. This enables you to interact with 
```

billy-the-fish commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -8,71 +8,110 @@ tags: [replicas, scaling, ha]
-up-to-date production data for analysis or to scale out reads beyond the limits of your primary data instance. You use 
-read replicas for read scaling. To limit data loss for your Timescale Cloud services, use [High availability][ha].
+A $READ_REPLICAs is a read-only copy of the primary data instance in your $CLOUD_LONG. Queries on $READ_REPLICAs have minimal impact on the performance of the primary data instance. This enables you to interact with 
+up-to-date production data for analysis or to scale out reads beyond the limits of your primary data instance. $READ_REPLICA_CAPs can be short-lived and deleted when a session of data analysis is complete, or long-running to power a
```
$READ_REPLICA_CAPs is not working

billy-the-fish commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -8,71 +8,110 @@ tags: [replicas, scaling, ha]
+You use $READ_REPLICAs for **read** scaling. To limit data loss for your $SERVICE_LONGs, use [$HA_REPLICAs][ha].
 
-## Create a read replica
+You create a $READ_REPLICA as a set of one or more nodes that share the same endpoint in $CONSOLE. You query each set as a single replica, and $CLOUD_LONG balances the load between the nodes within the set for you. 
```
```suggestion
A $READ_REPLICA is the set of one or more nodes that share the same endpoint in $CONSOLE. You query each set as a single replica. $CLOUD_LONG balances the load between the nodes in the set for you. 
```

billy-the-fish commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -8,71 +8,110 @@ tags: [replicas, scaling, ha]
 
-Read replicas can be short-lived and deleted when the analysis is complete, or long-running to power a
-business intelligence (BI) tool. To create a secure read replica for your read-intensive apps: 
+You can create as many $READ_REPLICAs as you need. For security and resource isolation, each $READ_REPLICA has unique connection details. To restrict access without isolation, create a [read-only role][read-only-role] for each $SERVICE_LONG. Users with read-only permissions cannot access the primary data instance directly.
```
Not sure about the link, best to have it in the prereqs only.

billy-the-fish commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -8,71 +8,110 @@ tags: [replicas, scaling, ha]
+- Create a target $SERVICE_LONG.
+- Create a [read-only user][read-only-role] on the primary data instance. 
+  
+   This user is propagated to the $READ_REPLICA when you create it.
```
```suggestion
   This user is propagated to the $READ_REPLICAs when you create them.
```

billy-the-fish reviewed (approved):
Few small nits. Good doc.

atovpeko committed:
Update use-timescale/ha-replicas/read-scaling.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Anastasiia Tovpeko <114177030+atovpeko@users.noreply.github.com>

atovpeko committed:
Update use-timescale/ha-replicas/read-scaling.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Anastasiia Tovpeko <114177030+atovpeko@users.noreply.github.com>

atovpeko committed:
Update use-timescale/ha-replicas/read-scaling.md

Co-authored-by: Iain Cox <iain@timescale.com>
Signed-off-by: Anastasiia Tovpeko <114177030+atovpeko@users.noreply.github.com>

atovpeko committed:
review

atovpeko committed:
review

atovpeko committed:
review

atovpeko committed:
review

atovpeko committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

rahilsondhi commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -8,71 +8,110 @@ tags: [replicas, scaling, ha]
-up-to-date production data for analysis or to scale out reads beyond the limits of your primary data instance. You use 
-read replicas for read scaling. To limit data loss for your Timescale Cloud services, use [High availability][ha].
+A $READ_REPLICA is a read-only copy of the primary data instance in your $SERVICE_LONG. Queries on $READ_REPLICAs have minimal impact on the performance of the primary data instance. This enables you to interact with 
+up-to-date production data for analysis or to scale out reads beyond the limits of your primary data instance. Read replicas can be short-lived and deleted when a session of data analysis is complete, or long-running to power a
+business intelligence (BI) tool. 
```
both use cases are about data analysis. maybe the second example could be about serving app servers instead?

```suggestion
or long-running to serve read traffic from your application servers.
```

billy-the-fish committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

billy-the-fish committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

ramonguiu commented:
I'll review in more detail but some high level feedback:
* I am not sure I would keep HA and Read Replicas in the same section. While replication is the technical solution for both, they are very different JTBD (jobs to be done). I would consider using Read scaling in the title instead of read replication.
* The name of the feature now is “read replica set” , not read replica.
* We have to mention horizontal scaling in the content.

billy-the-fish committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

atovpeko committed:
updates on review

atovpeko commented:
Hi @ramonguiu, thank you for reviewing! To address your comments:
- I have renamed the page as requested and mentioned horizontal scaling front and center. 
- This page will go under a big "Operate and scale" section during the upcoming doc restructure, so it won't be grouped with HA. 
- At the time of writing this, the designs still used both terms - read replica and read replica set - so I stuck to the old one. Updated now.
- Waiting when this is on dev to update the workflow, this is very outdated now. Will be happy to get your detailed review then.

atovpeko committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

billy-the-fish committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

Signed-off-by: Iain Cox <iain@timescale.com>

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,79 +1,115 @@
 
-You use read replicas to power your read-intensive apps and business intelligence tooling. Using read replicas to serve 
-reads for your app removes load from the primary data instance, and enables your service to improve ingest performance. 
+You use $READ_REPLICA sets in $CLOUD_LONG for horizontal read scaling, to power your read-intensive apps and business intelligence tooling. Using the replica sets to serve reads for your app unloads the primary data instance and enables your $SERVICE_SHORT to improve ingest performance. 
```
```suggestion
You use $READ_REPLICA sets in $CLOUD_LONG for horizontal read scaling, to power your read-intensive apps and business intelligence tooling. Additionally, using the replica sets to serve reads for your app unloads the primary data instance and enables your $SERVICE_SHORT to improve ingest performance. 
```

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,79 +1,115 @@
-replicas have minimal impact on the performance of the primary data instance. This enables you to interact with 
-up-to-date production data for analysis or to scale out reads beyond the limits of your primary data instance. You use 
-read replicas for read scaling. To limit data loss for your Timescale Cloud services, use [High availability][ha].
+A $READ_REPLICA is a read-only copy of your primary data instance. Queries on $READ_REPLICA have minimal impact on the performance of the primary instance. This enables you to interact with up-to-date production data for analysis, or to scale out reads beyond the limits of your primary instance. $READ_REPLICA_CAPs can be short-lived and deleted when a session of data analysis is complete, or long ...
```
```suggestion
A $READ_REPLICA is a read-only copy of your primary database instance. Queries on read replicas have minimal impact on the performance of the primary instance. This enables you to interact with up-to-date production data for analysis, or to scale out reads beyond the limits of your primary instance. $READ_REPLICA_CAPs can be short-lived and deleted when a session of data analysis is complete, or long-running to power an application or a business intelligence tool. 
```

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,79 +1,115 @@
-connection string to interact with each read replica. This provides both security and resource isolation. To restrict 
-access without isolation, you can create a [read-only role][read-only-role] for each Timescale Cloud service. Users 
-with read-only permissions cannot access the primary data instance directly.
+A $READ_REPLICA set in $CLOUD_LONG is a group of one or more $READ_REPLICA nodes that share the same endpoint. You query each set as a single replica. $CLOUD_LONG balances the load between the nodes in the set for you.
```
```suggestion
A $READ_REPLICA set in $CLOUD_LONG is a group of one or more $READ_REPLICA nodes that are accessed through the same endpoint. You query each set as a single replica. $CLOUD_LONG balances the load between the nodes in the set for you.
```

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,79 +1,115 @@
+- Create a target $SERVICE_LONG.
+- Create a [read-only user][read-only-role] on the primary data instance. 
+
+  A user with the read-only permissions cannot access the primary data instance directly. This user is propagated to the $READ_REPLICA set when you create them.
```
> A user with the read-only permissions cannot access the primary data instance directly.
This is not correct. All users are created in the primary and propagated to the replicas, including read-only users. A read-only user will not be able to make data changes on to the primary obviously.

atovpeko replied:
Updated

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,79 +1,115 @@
+- Create a target $SERVICE_LONG.
+- Create a [read-only user][read-only-role] on the primary data instance. 
+
+  A user with the read-only permissions cannot access the primary data instance directly. This user is propagated to the $READ_REPLICA set when you create them.
```
```suggestion
  A user with read-only permissions cannot make changes in the primary database. This user is propagated to the $READ_REPLICA set when you create them.
```

atovpeko replied:
Updated

ramonguiu commented:
I think we could add an architecture diagram to better illustrate the solution. There is one in the blog post (not sure if it's final): https://docs-dev.timescale.com/docs-297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling/use-timescale/297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling/ha-replicas/read-scaling/

atovpeko committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

atovpeko committed:
Update use-timescale/ha-replicas/read-scaling.md

Co-authored-by: Ramon Guiu <ramon@timescale.com>
Signed-off-by: Anastasiia Tovpeko <114177030+atovpeko@users.noreply.github.com>

atovpeko committed:
Update use-timescale/ha-replicas/read-scaling.md

Co-authored-by: Ramon Guiu <ramon@timescale.com>
Signed-off-by: Anastasiia Tovpeko <114177030+atovpeko@users.noreply.github.com>

atovpeko committed:
update for new UI

atovpeko committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

atovpeko committed:
update for new UI

atovpeko committed:
Merge remote-tracking branch 'origin/297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling

atovpeko committed:
diagram

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,78 +1,137 @@
+
+   The username and password of a read replica set are the same as the primary $SERVICE_SHORT. They cannot be changed independently. 
+
+   The connection information for each $READ_REPLICA set is unique. If you add or remove nodes from an existing set, the connection information of that set changes. To find the connection information for an existing $READ_REPLICA set: 
```
```suggestion
   The connection information for each $READ_REPLICA set is unique. You can add or remove nodes from an existing set and the connection information of that set will remain the same. To find the connection information for an existing $READ_REPLICA set: 
```

atovpeko replied:
@ramonguiu just checking: this was originally in the PRD.  I have tested this on dev and it seems the connection string did change when changing the number of nodes?

ramonguiu commented on use-timescale/ha-replicas/read-scaling.md:
```diff
@@ -1,78 +1,137 @@
+
+1. **Change the replica configuration and click `Apply`**
+
+   If you add or remove nodes from an existing $READ_REPLICA set, reconnect to the set using the updated connection details. Find the connection details 
```
```suggestion
```

ramonguiu committed:
Update use-timescale/ha-replicas/read-scaling.md

Signed-off-by: Ramon Guiu <ramon@timescale.com>

ramonguiu committed:
Update use-timescale/ha-replicas/read-scaling.md

Signed-off-by: Ramon Guiu <ramon@timescale.com>

billy-the-fish committed:
Merge branch 'latest' into 297-docs-rfc-new-docs-for-load-balancing-horizontal-scaling
